lib/http-bridge/test/integration/Test/Integration/Scenario/Wallets.hs

```diff
@@ -1020,6 +1020,252 @@ spec = do
+                   , Json [json| {
+                      "old_passphrase": 12345678910,
+                      "new_passphrase": "Secure passphrase"
+                         } |]
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "expected Text, encountered Number" ]
+                   )
+                 , ( "Num as new passphrase"
+                   , Json [json| {
+                      "old_passphrase": "Secure passphrase",
+                      "new_passphrase": 12345678910
+                         } |]
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "expected Text, encountered Number" ]
+                   )
+                 , ( "Missing old passphrase"
+                   , Json [json| {
+                      "new_passphrase": "Secure passphrase"
+                         } |]
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "key \"old_passphrase\" not present" ]
+                   )
+                 , ( "Missing new passphrase"
+                   , Json [json| {
+                      "old_passphrase": "Secure passphrase"
+                         } |]
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "key \"new_passphrase\" not present" ]
+                  )
+                ]
+        forM_ matrix $ \(title, payload, expectations) -> it title $ \ctx -> do
+            r <- request @ApiWallet ctx ("POST", "v2/wallets") Default simplePayload
+            let endpoint = "v2/wallets" </> (getFromResponse walletId r)
+                    </> ("passphrase" :: Text)
+            rup <- request @ApiWallet ctx ("PUT", endpoint) Default payload
+            verify rup expectations
+
+    it "WALLETS_UPDATE_PASS_04 - Deleted wallet is not available" $ \ctx -> do
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default simplePayload
+        let payload = updatePassPayload "Secure passphrase" "Secure passphrase2"
+        let delEndp = "v2/wallets" </> (getFromResponse walletId r)
+        _ <- request @ApiWallet ctx ("DELETE", delEndp) Default Empty
+
+        let updEndp = delEndp </> ("passphrase" :: Text)
+        rup <- request @ApiWallet ctx ("PUT", updEndp) Default payload
+        expectResponseCode @IO HTTP.status403 rup
+
+    describe "WALLETS_UPDATE_PASS_04 - non-existing wallets" $  do
+        forM_ falseWalletIds $ \(title, walId) -> it title $ \ctx -> do
+            let payload = updatePassPayload "Secure passphrase" "Secure passphrase2"
+            let endpoint = "v2/wallets" </> T.pack walId </> ("passphrase" :: Text)
+            rup <- request @ApiWallet ctx ("PUT", endpoint) Default payload
+            if (title == "40 chars hex") then
+                expectResponseCode @IO HTTP.status403 rup
+            else
+                expectResponseCode @IO HTTP.status404 rup
```

Comment:
There is a slight inconsistency in the behavior: There is `403` being returned in case when `{wallet-id}` is a 40 char long hex encoded string (ofc, for the wallet that does not exist). For any other string it returns `404` (e.g. longer/shorter than 40 chars, 40 char length but not valid hex). 

I think I'd be expecting either `404` for all. Or `400` for not valid IDs and `404` for valid and not existing (but this cannot be achieved now https://github.com/input-output-hk/cardano-wallet/pull/252#discussion_r282786569)

Not sure if this is something that can be addressed.

Reply:
Hmmm... Probably has to do with the way we map the error when a root private key can't be found. I'll have a look at it.